src/runtime/runtime.js

```diff
@@ -96,291 +83,418 @@
-   */
-  function isShimSymbol(symbol) {
-    return typeof symbol === 'object' && symbol instanceof SymbolValue;
+  var CONTINUATION_TYPE = Object.create(null);
+
+  function createContinuation(operand, thisArg, argsArray) {
+    return [CONTINUATION_TYPE, operand, thisArg, argsArray];
   }
 
-  function typeOf(v) {
-    if (isShimSymbol(v))
-      return 'symbol';
-    return typeof v;
+  function isContinuation(object) {
+    return object && object[0] === CONTINUATION_TYPE;
   }
 
-  /**
-   * Creates a new unique symbol object.
-   * @param {string=} string Optional string used for toString.
-   * @constructor
-   */
-  function Symbol(description) {
-    var value = new SymbolValue(description);
-    if (!(this instanceof Symbol))
-      return value;
+  var isTailRecursiveName = createPrivateName();
 
-    // new Symbol should throw.
-    //
-    // There are two ways to get a wrapper to a symbol. Either by doing
-    // Object(symbol) or call a non strict function using a symbol value as
-    // this. To correctly handle these two would require a lot of work for very
-    // little gain so we are not doing those at the moment.
-    throw new TypeError('Symbol cannot be new\'ed');
+  function initTailRecursiveFunction(func) {
+    func[isTailRecursiveName] = true;
+    return func;
   }
 
-  $defineProperty(Symbol.prototype, 'constructor', nonEnum(Symbol));
-  $defineProperty(Symbol.prototype, 'toString', method(function() {
-    var symbolValue = this[symbolDataProperty];
-    return symbolValue[symbolInternalProperty];
-    /* The implementation of toString below matches the spec, but prevents
-    use of Symbols in eg generators unless --symbol is set. To simplify our
-    code we deliberately go against the spec here.
-    if (!symbolValue)
-      throw TypeError('Conversion from symbol to string');
-    var desc = symbolValue[symbolDescriptionProperty];
-    if (desc === undefined)
-      desc = '';
-    return 'Symbol(' + desc + ')';
-    */
-  }));
-  $defineProperty(Symbol.prototype, 'valueOf', method(function() {
-    var symbolValue = this[symbolDataProperty];
-    if (!symbolValue)
-      throw TypeError('Conversion from symbol to string');
-    if (!getOption('symbols'))
-      return symbolValue[symbolInternalProperty];
-    return symbolValue;
-  }));
-
-  function SymbolValue(description) {
-    var key = newUniqueString();
-    $defineProperty(this, symbolDataProperty, {value: this});
-    $defineProperty(this, symbolInternalProperty, {value: key});
-    $defineProperty(this, symbolDescriptionProperty, {value: description});
-    freeze(this);
-    symbolValues[key] = this;
+  function isTailRecursive(func) {
+    return func[isTailRecursiveName] || false;
   }
-  $defineProperty(SymbolValue.prototype, 'constructor', nonEnum(Symbol));
-  $defineProperty(SymbolValue.prototype, 'toString', {
-    value: Symbol.prototype.toString,
-    enumerable: false
+
+  function tailCall(func, thisArg, argArray) {
```

Comment:
One of my motivations here was that we could make this part configurable. Not many people will use PTC so it would be good if they could skip this part of the runtime library.

Changing `$traceurRuntime.foo` to `$traceurRuntime$foo` is something I don't want to do at this point. We have a lot of work to do with our runtime and I'm not sure whether that will make things better or worse.